package dmPhysics;

import "ddf_extensions.proto";
import "ddf/ddf_math.proto";

option java_package = "com.dynamo.physics.proto";
option java_outer_classname = "Physics";
option (ddf_java_package) = "com.dynamo.physics.ddf";

message ConvexShape
{
    enum Type
    {
        BOX = 1;
        CAPSULE = 2;
        HULL = 3;
    }

    required Type  ShapeType = 1;

    /*
     ShapeType == BOX
     Data = [min_x, min_y, min_z, max_x, max_y, max_z]

     ShapeType == CAPSULE
     Data = [radius, height]

     ShapeType == HULL
     Data = [x0, y0, z0, x1, ...]
    */
    repeated float Data = 2;
}

message RigidBodyDesc
{
    required float Mass = 1;
    required string CollisionShape = 2;
}

message ApplyForceMessage
{
    required dmMath.Vector3 Force = 1;
    required dmMath.Point3 Position = 2;
}

message CollisionMessage
{
    required string OtherGameObjectId = 1;
}

message ContactPointMessage
{
    required dmMath.Point3 Position = 1;
    required dmMath.Vector3 Normal = 2;
    required string OtherGameObjectId = 3;
}